src/main/java/shop/service/impl/BookServiceImpl.java

```diff
@@ -20,23 +21,33 @@ public class BookServiceImpl implements BookService {
     @Override
     public BookDto createBook(CreateBookDto createBookDto) {
         return bookMapper.toBookDto(
-                bookRepository.createBook(
-                        bookMapper.toBook(createBookDto)
-                )
+                bookRepository.save(bookMapper.toBook(createBookDto))
         );
     }
 
     @Override
     public BookDto getBookById(Long id) {
-        Book bookFromRepository = bookRepository.findBookById(id).orElseThrow(
-                        () -> new EntityNotFoundException("Can`t get book by id: " + id));
-        return bookMapper.toBookDto(bookFromRepository);
+        return bookMapper.toBookDto(bookRepository.findById(id).get());
     }
 
     @Override
     public List<BookDto> getAllBooks() {
-        return bookRepository.findAllBooks().stream()
-                .map(bookMapper::toBookDto)
-                .toList();
+        return bookMapper.toBookDtoList(bookRepository.findAll());
+    }
+
+    @Override
+    public BookDto updateBookById(Long id, UpdateBookDto updateBookDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book not found. Id: " + id));
+        return bookMapper.toBookDto(
+                bookRepository.save(
+                        bookMapper.updateBookFromDto(updateBookDto, book)
+                )
```

Comment:
Let's improve readability
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Book not found. Id: " + id));
        bookMapper.updateBookFromDto(updateBookDto, book)
        return bookMapper.toBookDto(bookRepository.save(book));
```